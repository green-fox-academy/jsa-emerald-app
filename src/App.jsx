import React, { useState, useEffect } from 'react';
import { registerRootComponent, AppLoading } from 'expo';
import * as Font from 'expo-font';
import { Provider } from 'react-redux';
import { createAppContainer } from 'react-navigation';
import { Alert } from 'react-native';
import { PersistGate } from 'redux-persist/integration/react';
import AppNavigator from './Navigator';
import { store, persistor } from './reduxStore';
import Roboto from '../assets/Fonts/Roboto/Roboto-Regular.ttf';
import RobotoMedium from '../assets/Fonts/Roboto/Roboto-Medium.ttf';

const AppContainer = createAppContainer(AppNavigator);

const App = () => {
  const [fontLoading, setFontLoading] = useState(true);

  useEffect(() => {
    async function loadFont() {
      try {
        await Font.loadAsync({
          Roboto,
          Roboto_medium: RobotoMedium,
        });
        setFontLoading(false);
      } catch (err) {
        Alert.alert(
          'Failed to load custom Font',
        );
      }
    }
    loadFont();
  }, []);

  return fontLoading ? <AppLoading />
    : (
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <AppContainer />
        </PersistGate>
      </Provider>
    );
};

registerRootComponent(App);
