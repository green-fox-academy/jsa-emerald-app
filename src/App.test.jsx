import React from 'react';
import App from './App';

describe('<App />', () => {
  it('render', () => {
    const tree = JSON.stringify(<App />);
    expect(tree).toMatchSnapshot();
  });
});
