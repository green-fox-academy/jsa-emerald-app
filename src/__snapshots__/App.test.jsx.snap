// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render 1`] = `"{\\"type\\":{},\\"key\\":null,\\"ref\\":null,\\"props\\":{},\\"_owner\\":null,\\"_store\\":{}}"`;
